drivers/display/display_framebuf.c

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2019 Intel Corp.
+ *
+ * This is most of the display driver for a "standard" 32-bpp framebuffer.
+ * Device-specific drivers must still create the device instance and initialize
+ * it accordingly, but this driver implements most/all of the API functions
+ * for simple ARGB and monochrome modes.
+ *
+ * This code attempts to be endian-agnostic. It manipulates the framebuffer
+ * address space only in 32-bit words (and assumes those words are 0xAARRGGBB).
+ * Similarly, it will manipulate display buffers with smaller color spaces only
+ * in their natural word sizes. If applications adhere to the same rules, they
+ * should encounter no trouble. Avoid manipulating image data as byte arrays.
+ *
+ * Note that this implementation isn't particularly fast; it's aimed at
+ * simplicity. There's a lot of room for performance improvement.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <display.h>
+#include <display/framebuf.h>
+#include <string.h>
+
+#define GREY 0x00C0C0C0
+#define BLACK 0x00000000
+
+static int framebuf_blanking_on(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static int framebuf_blanking_off(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static void *framebuf_get_framebuffer(const struct device *dev)
```

Comment:
I would expect a direct pointer to the framebuffer is returned by this function, in this case the upper layers could directly draw into to the frame buffer memory.

Reply:
I was actually going to propose that this API be removed entirely, as I can see no point in the display drivers if the client is simply going to write directly on the framebuffer.

Reply:
Frame buffer pointer could be provided by target specific hal layer, this API provides a common interface to get access to such pointers.

Reply:
I've left this out, still, because there is no API mechanism to retrieve the pitch of the framebuffer returned by this pointer. I will put in another PR to add such a mechanism and cycle back and add this API to this driver once that's settled.